src/rules/noAssignmentExpression.ts

```diff
@@ -17,7 +18,21 @@ export class NoAssignmentExpression implements SourceError {
   }
 
   public elaborate() {
-    return ''
+    function lastAssignmentNode(theNode: es.AssignmentExpression): es.AssignmentExpression {
+      if (theNode.right.type === 'AssignmentExpression') {
+        return lastAssignmentNode(theNode.right)
+      } else {
+        return theNode
+      }
+    }
+
+    const lastNode = lastAssignmentNode(this.node)
+    const leftStr = generate(this.node.left)
+    const rightStr = generate(lastNode.right)
+
+    const elabStr = `Try moving this to another line:\n\n\t${leftStr} = ${rightStr};`
```

Comment:
Just thought about something. 

Would it be feasible to check if this assignment is in an IfStatement, WhileStatement, ForStatement, LogicalExpression, or ConditionalExpression test condition?

Then we could say something like the multiplication thing, 

If you were planning to check if left is equal to right, use === instead.

Reply:
I don't think that's relevant? Correct me if I'm wrong, but this rule is only broken by code that looks like 
`let a = 1; 
let b = a = 2;`
Whereas your equivalence checking is more likely to occur from statements like 
`c = 5`

Reply:
I mean it's also broken in places like
```js
if (a = b) { //meaning to check if a === b
  dostuff();
} else {}
```

Reply:
In any case, this can be done in another PR.